### PR TITLE
Refactor getting region clipping rectangles

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Two commits, the first is just a style cleanup for a section of DataGridView around the consuming code for this change.

The second unifies RGNDATAHEADER helper code in the primitives assembly and makes a minor tweak for a test where I first implemented the newer, better perf helper.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/4013)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.Rendering.cs [hunk 1]
```diff
@@ -23,9 +23,8 @@ public void ToolStrip_RendersBackgroundCorrectly()
             form.Controls.Add(toolStrip);
 
             // Force the handle creation
-            _ = form.Handle;
-            _ = toolStrip.Handle;
-            form.PerformLayout();
+            Assert.False(form.Handle == IntPtr.Zero);
+            Assert.False(toolStrip.Handle == IntPtr.Zero);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.Rendering.cs:27]:
```suggestion
            Assert.NotEqual(IntPtr.Zero, form.Handle);
            Assert.NotEqual(IntPtr.Zero, toolStrip.Handle);
```

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs [hunk 2]
```diff
@@ -3104,60 +3032,48 @@ internal bool IsCurrentRowDirtyInternal
...
-        }
+        private bool IsEscapeKeyEffective => _dataGridViewOper[OperationTrackColResize]
+            || _dataGridViewOper[OperationTrackRowResize]
+            || _dataGridViewOper[OperationTrackColHeadersResize]
+            || _dataGridViewOper[OperationTrackRowHeadersResize]
+            || _dataGridViewOper[OperationTrackColRelocation]
+            || IsCurrentCellDirty
+            || ((VirtualMode || DataSource != null) && IsCurrentRowDirty)
+            || (EditMode != DataGridViewEditMode.EditOnEnter && EditingControl != null)
+            || _dataGridViewState1[State1_NewRowEdited];
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs:3043]: ❗ This is a functional change, it would be better to have it as a separate commit.

**Review by RussKie — CHANGES_REQUESTED** [review]

File: src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs (hunk 2 above)
Comment (JeremyKuhne) [reply src/System.Windows.Forms/src/System/Windows/Forms/DataGridView.cs:3043]: It isn't a functional change, this is all tool done.

**Review by RussKie — APPROVED** [review]

**Review by RussKie — APPROVED** [review]
